pipeline/clean/opuscleaner/generate_filters.py

```diff
@@ -0,0 +1,78 @@
+"""
+Generates filter config for a dataset based on defaults to use in OpusCleaner
+"""
+
+import argparse
+import json
+import os
+from typing import Dict, Optional
+
+
+def find_custom_filter(src: str, trg: str, dataset: str) -> Optional[Dict]:
+    # TODO: we'll likely need to move to a separate repo for those
+    # TODO: to not include all filters for all languages in TC artifacts
+
+    # workaround: we use "_" to separate the dataset version for OPUS datasets and OpusCleaner uses "-"
+    idx = dataset.rfind("_")
+    dataset_opus = f"{dataset[:idx]}-{dataset[idx + 1:]}" if idx else ""
+
+    paths = [
+        f"configs/{src}-{trg}/{dataset}.{src}-{trg}.filters.json",
+        f"configs/{trg}-{src}/{dataset}.{trg}-{src}.filters.json",
+        f"configs/{trg}-{src}/{dataset_opus}.{trg}-{src}.filters.json",
+        f"configs/{src}-{trg}/{dataset_opus}.{src}-{trg}.filters.json",
+    ]
+
+    for path in paths:
+        if os.path.exists(path):
+            with open(path) as f:
+                config = json.load(f)
+                print(f"Found custom filter {path}")
+                return config
+
+    return None
+
+
+def build_default_config(src: str, trg: str) -> str:
+    # note: do not call the folder with default filters "filters" because it's a magic word for opuscleaner-clean
+    # and it starts processing such folder
+    # TODO: ideally "other" for "deescape-special-chars" should be replaced to <trg> for supported languages
+    with open("configs/default.filters.json") as f:
+        config_str = f.read()
+        config_str = config_str.replace("<src>", src).replace("<trg>", trg)
+        abs_path_patterns = os.path.abspath("configs/remove_frequent_patterns.txt")
+        config_str = config_str.replace("configs/remove_frequent_patterns.txt", abs_path_patterns)
+        return json.loads(config_str)
+
+
+def main() -> None:
```

Comment:
Suggestion: A test for this would be good to have. It would be nice to run a test exercising the CLI arguments. We don't have the first test written using pytest, so the burden is a bit higher. If you don't want to write a test for it, then I could potentially follow-up and write a test for it after my single process work.

I would imagine we would have some kind of test utilities where we could run actual commands like

```py
test_generate_filters():
     results = run_cli('pipeline/clean/opuscleaner/generate_filters.py --src="en" --trg="ru"...')
     assert ...
````

Reply:
We'll need to set up this first. I filed an issue https://github.com/mozilla/firefox-translations-training/issues/209